### PR TITLE
felix/bpf: fix err handling in bpf_program_attach_cgroup

### PR DESCRIPTION
**Pull request by tomastigera**
As the error is encoded in the link pointer, it it not NULL in case of
an error.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by sridhartigera — APPROVED** [review]
LGTM

File: felix/bpf/libbpf/libbpf_api.h [hunk 1]
```diff
@@ -136,11 +136,13 @@ struct bpf_link *bpf_program_attach_cgroup(struct bpf_object *obj, int cgroup_fd
 
 	if (!(prog = bpf_object__find_program_by_name(obj, name))) {
 		err = ENOENT;
+		link = ERR_PTR(-ENOENT);
```
Comment (fasaxc) [inline felix/bpf/libbpf/libbpf_api.h:139]: Given that we're using errno, should we just set link to NULL if errno is non-zero?  That'd match Go's normal convention

Comment (tomastigera) [reply felix/bpf/libbpf/libbpf_api.h]: Yup, we got confused when I set link=NULL and then it kept failing, but the issue is that we do not rebuild when the cgo headers change. So fixing both.

**Pull request merged**